engine/src/mbliphonedc.mm

```diff
@@ -1184,18 +1184,59 @@ void MCIPhoneRunBlockOnMainFiber(void (^block)(void))
 // MW-2012-08-06: [[ Fibers ]] Updated entry point for didBecomeActive.
 static void MCIPhoneDoDidBecomeActive(void *)
 {
-	extern char **environ;
-	char **env;
-	env = environ;
-	
 	// MW-2011-08-11: [[ Bug 9671 ]] Make sure we initialize MCstackbottom.
 	int i;
 	MCstackbottom = (char *)&i;
 	
 	NSAutoreleasePool *t_pool;
 	t_pool = [[NSAutoreleasePool alloc] init];
-	
-	char *args[1];
+
+    // SN-2015-09-22: [[ Bug 15987 ]] Use NSProcessInfo to get the env variables
+    char **env;
+    uindex_t envc;
+    
+    envc = (uindex_t)[[[NSProcessInfo processInfo] environment] count];
+    env = NULL;
+    
+    // last elem of env must be a NULL element, so we allocated envc + 1 elems
+    if (MCMemoryNewArray(envc + 1, env))
+    {
+        NSDictionary *t_environment;
+        NSEnumerator* t_key_enumerator;
+        NSString* t_key;
+        NSString* t_value;
+        
+        // NSProcessInfo::environment returns NSDictionary<NSString*,NSString*>
+        t_environment = [[NSProcessInfo processInfo] environment];
+        
+        t_key_enumerator = [t_environment keyEnumerator];
+        uindex_t t_index = 0;
+        
+        // Loop through all the keys in the environment array
+        while (t_key = (NSString*)[t_key_enumerator nextObject])
```

Comment:
It might be worth updating this to use Obj-C iterator syntax rather than an explicit while.
```
for(id key in myDict) {
    id value = [myDict objectForKey:key];
    [value doStuff];
}
```